src/main/java/org/spongepowered/api/util/TickBasedByDefault.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Used to annotate time based types that they are tick based. This means
+ * that e.g. a duration of 5 seconds will be ideally 100 minecraft ticks.
+ * But in reality can 100 ticks be longer then 5 seconds depending on the
+ * actual tick length.
+ *
+ * <p>You can't assume that the duration of a given length will be exactly
+ * the same, this depends on the implementation.</p>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE_USE)
+public @interface TickBasedByDefault {
```

Comment:
If the whole point of this PR is to remove any notion of a "tick", this name is screaming of "our reference implementations use ticks". It does, but that's an implmentation detail and doesn't really explain why that's important. I also know we're not going to get away from talking about ticks completely - but I think references to ticks should be consigned to the javadocs.

What you're _really_ saying here is that the input is the minimum time - if the server is overloaded the target may take longer than indicated. `@PerformanceDependent` might be a better name, or just some name that indicates that a lagging server may extend the time, rather than stating that we're just estimating based on tick time. That way we're stating that a lagging server may extend the duration, without suggesting that we're backing the field with ticks, and we're not expressing an implementation detail.

This PR is also a break from working with ticks for a lot of developers. If we're removing ticks, then this needs to be crystal clear as to what these durations mean. I think the javadoc needs rewording, something like (but perhaps not exact):

> Marks a parameter or return type that represents a duration as dependent on server performance, and should be considred a minimum time, rather than an absolute time.
>
> If the server operates under perfect conditions (for example, 20 ticks per second (TPS) in vanilla minecraft), then the duration that has been set or returned will be accurate. However, if the server performance is degraded at any moment (in vanilla, a drop in the TPS), then the given duration might be considered an underestimate if the underlying implementation uses ticks rather than wall clock time to time actions. 
>
> The effect of server performance drops on annotated types is dependent on the implementation that is being used and the settings that are applied. Therefore, no assumptions should be made about how durations are calculated, it is implementation dependent. 
>
> Plugins wishing to execute tasks based on any returned duration annotated with this annotation should check that the desired action has occured rather than solely relying on what is returned. Plugins setting times annotated with this annotation should be aware that the underlying implementation may convert the time into a different format and, as such, the stored time may not be exactly the same as requested.
>
> Any duration annotated with this annotation may be used with any of the units provided in `{@link TemporalUnits}`.